src/server/config/complete.js

```diff
@@ -20,13 +20,23 @@
 import { difference } from 'lodash';
 import { transformDeprecations } from './transform_deprecations';
 import { unset, formatListAsProse, getFlattenedObject } from '../../utils';
+import { createTransform, Deprecations } from '../../deprecation';
+
 
 const getFlattenedKeys = object => (
   Object.keys(getFlattenedObject(object))
 );
 
-const getUnusedConfigKeys = (disabledPluginSpecs, rawSettings, configValues) => {
-  const settings = transformDeprecations(rawSettings);
+const getUnusedConfigKeys = (plugins, disabledPluginSpecs, rawSettings, configValues) => {
+  // transform deprecated core settings
+  let settings = transformDeprecations(rawSettings);
+
+  // transform deprecated plugin settings
+  plugins.forEach(({ spec }) => {
+    const deprecationsProvider = spec.getDeprecationsProvider();
+    if (!deprecationsProvider) return;
+    settings = createTransform(deprecationsProvider(Deprecations))(settings);
```

Comment:
According to `src/plugin_discovery/plugin_config/settings.js` deprecationsProviders can be async, so we should handle that here.